### PR TITLE
update codeModel in the Second-phase async init

### PR DESCRIPTION
**Pull request by elahehrashedi**
bug fix: https://github.com/microsoft/vscode-cmake-tools/issues/2915

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -174,6 +174,8 @@ export class ExtensionManager implements vscode.Disposable {
             }
             await this.initActiveProject();
         }
+        rollbar.takePromise('Post-folder-open', { folder: this.getActiveProject()?.folderPath }, this.postWorkspaceOpen(this.getActiveProject()));
```
Comment (bobbrow) [inline src/extension.ts:177]: It doesn't seem like this one is needed.  `postUpdateActiveProject` is called before we get here so `postWorkspaceOpen` will have already been called.

**Review by bobbrow — CHANGES_REQUESTED** [review]
Please test without the change in `init`.  I don't think it's needed.

**Comment by bobbrow** [comment]
> It doesn't seem like this one is needed. `postUpdateActiveProject` is called before we get here so `postWorkspaceOpen` will have already been called.

Or perhaps what we need to do instead is call `postWorkspaceOpen` on all projects, not just the active one.  That's what we used to do.  I see that the project outline does not initialize all the projects when you first open the workspace.  This is because we don't configure them all.  This could be considered a regression.

**Comment by bobbrow** [comment]
`postWorkspaceOpen` should only ever be called once per project. I changed the logic and pushed a change which appears to work for my multi-project workspace.  Please confirm that this works for you as well.

**Comment by elahehrashedi** [comment]
In my vc multi-project, this worked after a re-load, the first time that I opened the project, the browse path was still not available.
I have another simpler multi-project that this worked on.

**Review by bobbrow — APPROVED** [review]

**Pull request merged**